nanofi/CMakeLists.txt

```diff
@@ -23,32 +23,67 @@ IF(POLICY CMP0048)
   CMAKE_POLICY(SET CMP0048 OLD)
 ENDIF(POLICY CMP0048)
 
-include_directories(include)
-include_directories(../libminifi/include ../thirdparty/cron ../thirdparty/spdlog-20170710/include)
-include_directories(../thirdparty/ut)
+include(ExternalProject)
+
+if(MSVC)
+    set(CBOR_RESTRICT_SPECIFIER "")
+elseif(APPLE)
+    set(CBOR_RESTRICT_SPECIFIER "restrict")
+else()
+    set(CBOR_RESTRICT_SPECIFIER "__restrict__")
+endif()
+
+ExternalProject_Add(
+    libcbor-external
```

Comment:
What are we using CBOR for? I see that it is being used for encoding C2 protocol payloads, but as far as I know our CoAP C2 payload is a custom format, not CBOR.
What have you tested this with?

Reply:
I have tested this with coap_server.c and c2_client.c in nanofi/ecu.
The coap_server.c stands as a C2 server listening for heartbeats from ecus. c2_client.c stands as a c2 client requesting operations on ecu via coap_server.

I will write more unit tests for testing the serde of c2 protocol including heartbeats and c2 operations.